droidlet/dashboard/web/src/components/Interact/Question.js

```diff
@@ -1,7 +1,27 @@
 /*
    Copyright (c) Facebook, Inc. and its affiliates.
 
- * Question.js handles the template for the question and answers in NSP error labeling
+ * Question.js handles the template for the question and answers in error labeling
+ * The error labeling logical tree is as follows:
+                      Outcome As Expected?
+                             /   \
+               No Error <- Yes    No
+                                  |
+                        Did Agent Understand?
+                               /   \
+                             Yes    No -> NSP Error
+                              |
+                    Location Ref Exists?
+                            /   \
+                          Yes    No -> Other Error
+                           |
+             Found Location Ref in Memory?
+                         /   \
+                       Yes    No -> Perception Error
+                        |
+           Is 'This' the Location Ref?
```

Comment:
Maybe for clarity "Location Ref" -> "Reference Obj" since it doesn't strictly need to be in the Location subcomponent?

Reply:
Right yes, I did not update the comments yet, will do once I have a sign off on the implementation